clang/lib/CIR/Lowering/DirectToLLVM/LowerAttrToLLVMIR.cpp

```diff
@@ -34,6 +37,23 @@ class CIRDialectLLVMIRTranslationInterface
   amendOperation(mlir::Operation *op, mlir::NamedAttribute attribute,
                  mlir::LLVM::ModuleTranslation &moduleTranslation) const final {
     // TODO: Implement this
+    auto func = dyn_cast<mlir::LLVM::LLVMFuncOp>(op);
+    if (!func)
+      return mlir::success();
+    llvm::Function *llvmFunc = moduleTranslation.lookupFunction(func.getName());
+    if (auto extraAttr = attribute.getValue()
+                             .dyn_cast<mlir::cir::ExtraFuncAttributesAttr>()) {
+      for (auto attr : extraAttr.getElements()) {
+        if (auto inlineAttr = attr.getValue().dyn_cast<mlir::cir::InlineAttr>()) {
+          if (inlineAttr.isNoInline())
+            llvmFunc->addFnAttr(llvm::Attribute::NoInline);
+          else if (inlineAttr.isAlwaysInline())
+            llvmFunc->addFnAttr(llvm::Attribute::AlwaysInline);
+          else if (inlineAttr.isInlineHint())
+            llvmFunc->addFnAttr(llvm::Attribute::InlineHint);
```

Comment:
add a `else` and add an assert, in case we add more enums this shouldn't silently miss it. Alternatively you could use a switch on the enums as well, if you want.